[TableGen] Fix a potential crash when operand doesn't appear in the instruction pattern

shiltian opened the pull request:
We have a check of whether an operand is in the instruction pattern, and emit an
error if it is not, but we simply continue execution, including directly
dereferencing a point-like object `InVal`, which will be just created when
accessing the map. It contains a `nullptr` so dereferencing it causes crash.
This is a very trivial fix.

kosarev commented on llvm/utils/TableGen/Common/CodeGenDAGPatterns.cpp:
```diff
@@ -3872,6 +3872,7 @@ void CodeGenDAGPatterns::parseInstructionPattern(CodeGenInstruction &CGI,
       }
       I.error("Operand $" + OpName +
               " does not appear in the instruction pattern");
+      continue;
```
Should we also `continue` on the empty-name error above?

kosarev reviewed (approved):
LGTM.

shiltian committed:
[TableGen] Fix a potential crash when operand doesn't appear in the instruction pattern

We have a check of whether an operand is in the instruction pattern, and emit an
error if it is not, but we simply continue execution, including directly
dereferencing a point-like object `InVal`, which will be just created when
accessing the map. It contains a `nullptr` so dereferencing it causes crash.
This is a very trivial fix.
